share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -107,6 +107,87 @@ function(sysbuild_get variable)
   endif()
 endfunction()
 
+# Usage:
+#   sysbuild_cache(CREATE APPLICATION <name> [CMAKE_RERUN])
```

Comment:
How should this relate to `sysbuild_cache_set()`? Also, will there be other modes besides `CREATE`?

Reply:
guess we're getting into a naming scope issue here.
`sysbuild_cache_set()` is setting a value directly in the CMake cache of an image, that is `<image-bin>/CMakeCache.txt`.
But because it's a sysbuild specific extension then that function is prefixed with `sysbuild_`.

`sysbuild_cache()` is a function working on the `sysbuild` cache of an image, that is `<sysbuild-bin>/<image>_sysbuild_cache.txt`.
So if namespacing that function the same way, then it ought to be called `sysbuild_sysbuild_cache()`.
Kind of confusing naming.

I'm open to alternate function name proposals.

Reply:
> `sysbuild_cache_set()` is setting a value directly in the CMake cache of an image, that is `<image-bin>/CMakeCache.txt`.

That's what the description says, but on second glance, this doesn't match what the function does. It's an extension of `set(... CACHE ...)`, so it actually just updates values in `<sysbuild-bin>/CMakeCache.txt`. It doesn't touch any image's CMake cache or sysbuild cache by itself.

Hence, `sysbuild_cache_set()` isn't specific to sysbuild. We could move it to `extensions.cmake` and call it `zephyr_set_cache()`, which would resolve the namespace conflict. Not blocking this PR though.

I think `sysbuild_cache()` can keep its name as is. We should just be more careful not to mix up the concepts of a sysbuild cache and sysbuild's own CMake cache.

Reply:
> That's what the description says, but on second glance, this doesn't match what the function does. It's an extension of set(... CACHE ...), so it actually just updates values in <sysbuild-bin>/CMakeCache.txt. It doesn't touch any image's CMake cache or sysbuild cache by itself.

Not directly, that's true.
But it creates an entry in sysbuild CMake cache which is propagated to the `<image>_sysbuild_cache.txt`.

But on closer look at how the actual implementation has been done, then it does appear that this could be moved the more generic Zephyr `extensions.cmake`.